### PR TITLE
Registration related fixes

### PR DESCRIPTION
**Pull request by RusiPapazov**
Migrating changes done in https://github.com/clippings/clippings/pull/7458

- Extract `FormGroup` from `TextInput`
- Extract `FormError` from `TextInput`
- Extract `Label` from `TextInput`
- Add forwardRef to `TextInput`
- Add forwardRef to `Select`
- Add max-height to `Select`
- Use `text-200` for form texts

### CODE REVIEW

File: src/components/Select/types/SelectPropsType.ts [hunk 1]
```diff
@@ -7,5 +7,6 @@ export type SelectPropsType = {
   }[];
   selected?: any | number | string;
   placeholder?: string;
-  onChange: Function;
+  onChange?: Function;
```
Comment (T3ch0o) [inline src/components/Select/types/SelectPropsType.ts:10]: This has to be updated with (value: string) => void I assume

Comment (RusiPapazov) [reply src/components/Select/types/SelectPropsType.ts:10]: Will do.

**Comment by T3ch0o** [comment]
One more thing on the for the prop types we add PropsType.ts at the end just to follow the naming convention lets follow it.

**Comment by RusiPapazov** [comment]
> One more thing on the for the prop types we add PropsType.ts at the end just to follow the naming convention lets follow it.

@T3ch0o Noticed it, but it doesn't seem consistent right now. Will update what I changed though.

**Review by T3ch0o — APPROVED** [review]

File: src/stories/Select/Select.stories.tsx [hunk 2]
```diff
@@ -18,13 +18,16 @@ const options = [
   },
 ];
 
+const ref = React.createRef();
+
 const selectInput = () => {
   const placeholder = text('Placeholder', 'Select a value');
   const onChange = value => {
+    console.log(ref.current);
```
Comment (MartinAndreev) [inline src/stories/Select/Select.stories.tsx:26]: Is this intentional?

File: src/stories/TextInput/TextInput.stories.tsx [hunk 3]
```diff
@@ -5,15 +5,20 @@ import { TextInput } from '@paper/components';
...
 
+const ref = React.createRef();
+
 const baseProps = {
+  id: 'text-input',
   title: 'Label',
   placeholderText: 'Placeholder',
   disabled: false,
   onChange: () => {
+    console.log(ref.current);
```
Comment (MartinAndreev) [inline src/stories/TextInput/TextInput.stories.tsx:16]: Same here.

File: src/stories/Select/Select.stories.tsx (hunk 2 above)
Comment (RusiPapazov) [reply src/stories/Select/Select.stories.tsx:26]: Yes, to show the ref. Any suggestion as to how to show, or remove it at all is welcome.

Comment (MartinAndreev) [reply src/stories/Select/Select.stories.tsx:26]: Its ok in the stories, but have we installed the console extension to storybook?

https://www.npmjs.com/package/@storybook/addon-console

Comment (RusiPapazov) [reply src/stories/Select/Select.stories.tsx:26]: I can give it a try

**Review by MartinAndreev — APPROVED** [review]

**Pull request merged**